### DISCUSSION TITLE
Repeating a logic state - help please

### DISCUSSION THREAD

**Discussion by zoelang30** [bonsai-rx/bonsai#1928] [Q&A]
Hello, 
There is most likely a very basic solution to my problem. For context: I have a UV LED strip controlled by a teensy using the SendSysex commands in Bonsai. The UV strip will either have the top section ON or the bottom section ON based on ROI activation at the top or bottom of the arena. 

When the bottom ROI is activated the top section of the LED strip will go on and remain on until the top ROI is activated which will the turn off the top section of the strip and turn on the bottom. I need this loop to be continuous until the experiment is ended by me. 

My bonsai script works for the first activation of each ROI but then does not repeat. 
I have tried using the **repeat** node at the end of my state sequence, but it still does not repeat. 

Does anyone have a suggestion on how to ensure this logic continues to loop?

Please see script attached. 
Thank you!

[UVstimulus - editing.bonsai.zip](https://github.com/user-attachments/files/16284128/UVstimulus.-.editing.bonsai.zip)

**Comment by bruno-f-cruz** [comment]
Hey,

Can you give me a few more hints of where in the workflow should i be looking at? There are no repeaet nodes.

Thanks

**Reply by zoelang30** [reply]
Hi Bruno, 

I tried the repeat node at the end of the of this branch:
![image](https://github.com/user-attachments/assets/a8b22ae4-81d1-4290-ac21-b3d0b08042d2)

I have had some help to instead change the branches from the **ROI activity detected** node to change from a boolean value to a byte value using C#. This seems to be working and doing what it should. Workflow also attached.
![image](https://github.com/user-attachments/assets/56e9e0e8-568a-4547-9980-4f3817bc61cc)
[UVstimulus - editing - bytearraywrapper.bonsai.zip](https://github.com/user-attachments/files/16286514/UVstimulus.-.editing.-.bytearraywrapper.bonsai.zip)
[UVstimulus - editing.bonsai.zip](https://github.com/user-attachments/files/16286515/UVstimulus.-.editing.bonsai.zip)

Thank you for the quick response though!
